project_management/estimating.md

```diff
@@ -0,0 +1,65 @@
+# Estimating
+
+Estimates are key to what we do.
+
+We are in the business of selling software creation hours. We want to do this as inexpensively as possible while maintaining a healthy revenue for growth and downtimes. As a group, we need to hold appropriate tension between cost and quality.
+
+Estimates should be made by two developers at a minimum; estimates are shared, and ideally by the implementing team.
+When you win and lose together, your estimates get even better.
+
+## Derive duration
+
+We focus on size estimates (how much), not duration estimates (how long). Related, but useful independently.
+Estimates are a combination of effort, risk, complexity, etc. Separating size from duration enables agility in scheduling and iteration planning.
+
+## Split the work
+
+Before the work can be estimated, it has to be split into chunks of a manageable size. Depending on how reliable it needs to be (the estimate type), we will create user stories encompassing a larger amount of functionality or more detailed user stories that define a specific functionality. The team must focus on providing just enough details so that the client can make the right decision.
+
+## T-shirt sizes
+
+T-shirt sizes is the method we use to estimate software size.
+
+The estimating team:
+
+1. **Take one user story at a time.**
+2. **Ensure that everyone involved knows what the end result will be like once the story is completed.** To achieve this, the team will discuss the feature until they have just enough common understanding to give a reasonable enough estimate. This could include what is not included in the story, different possibilities of implementation, concerns, assumptions, or ideas.
+3. **Share their estimate (planning poker) as a t-shirt size** XS, S, M, L, or XL. XL is commonly referred as "a story that needs to be split before we can give an accurate enough estimate". XS is usually reserved for specifically small features.
+4. **Assign a certainty level.** They make the difference between an optimistic scenario and the worst scenario. The certainty levels will multiply the best case scenario estimate by a factor to yield the worst case scenario. E.g. certainty 3 (uncertain) may have a factor of 4. Therefore, an M estimate defined as 2 pair days may take between 2 and 8 pair days of work.
```

Comment:
I'm not sure I get this calculation.
Certainty 3 means 4, which is why 2 becomes 8?

Reply:
I understand the confusion. Maybe we should describe levels of uncertainty with "low, medium, high" instead of numbers as to avoid confusing different types of numbers. It would also be useful to indicate that factors go both ways, so a factor 4 means that the estimate could potentially be multiplied or divided by 4. 

I find the Cone of Uncertainty to be a very useful framework for having this conversation: http://www.agilenutshell.com/cone_of_uncertainty

Reply:
Excellent suggestion. I've changed the template. I'll also make sure it's reflected in the document.